pengine/notif.c

```diff
@@ -103,7 +103,9 @@ expand_node_list(GListPtr list, char **uname, char **metal)
         int existing_len = 0;
         node_t *node = (node_t *) gIter->data;
 
-        CRM_ASSERT(node->details->uname);
+        if (node->details->uname == NULL) {
+            continue;
```

Comment:
If there is no uname, then some placeholder such as "NA" or "unknown" needs to be used instead.
This index of nodes between the uname and physical host name needs to match exactly.  _Everything_ depends on that.

Reply:
expand_node_list() is only used for notify_available_uname, notify_all_uname and notify_all_hosts -- none of those are matched up with the others iiuc

expand_list() is used for the others

Reply:
Ohhh, it won't appear in notify_all_uname OR notify_all_hosts if there is no uname.  Thats fine.